collector.go

```diff
@@ -61,6 +61,10 @@ func NewCollector(config CollectorConfig) (*Collector, error) {
 	return c, nil
 }
 
+func (c *Collector) SetEndpoints(endpoints []*url.URL) {
+	c.endpoints = endpoints
+}
```

Comment:
I think this will not be reliable. The collector works asynchronously which makes a mutex necessary to always gather consistent data. When we modify the endpoints while the collector iterates over this list we might cause indeterministic behaviour. The most simple thing would be to initialize a new collector with the appropriate endpoints. If this has to be dynamic on a single collector instance, then we have to ensure the collector works as expected when changing its endpoints. We need then unit tests for this and lock the setter here using the mutex. Additionally in `Collect` we need to change the locking so that the whole iteration is locked and not only the response setting.

Reply:
This is reliable when used synchronously with the `Collect` method.
Initializing a new collector is not something I can do in my case because it is shared in different places.
I don't want to spend time implementing this entire mutex thing when it is not needed now.

Reply:
I could add a comment to warn that this function is not thread-safe, would this be ok ?

Reply:
I just verified this myself to understand how it works on go. The value which is iterate on it captured when the loop starts. So we should be fine. Just add a comment to make this clear in the godoc. Thank you.

Reply:
Just verified the same things. When using range the value is read once when the loop starts.
https://play.golang.com/p/7elLaWEYWtM